packages/cloud-functions/src/cico/fetchProviders.ts

```diff
@@ -48,7 +59,7 @@ export const fetchProviders = functions.https.onRequest(async (request, response
     XANPOOL_RESTRICTED,
   } = getProviderAvailability(requestData.userLocation)
 
-  const [simplexQuote] = await Promise.all([
+  const [simplexQuote, moonpayQuote, xanpoolQuote, transakQuote] = await Promise.all([
```

Comment:
Since this is making all these requests we're making users wait seeing a loading screen until the slowest one returns.

This is fine for now, but something we could do is either have this one return the providers and the client make N more requests (one per provider) to get the individual fees while showing a loading only in the providers that are still loading.

There might be a better solution as well, but this was the first thing that came to mind.

Reply:
That's how we had it before but the design decision was to show the loading indicator until they are all done. Since we wanted the design to be like that, we merged all the requests into one endpoint and have a 15 sec timeout on each fetch for providers. If we start seeing endpoints taking a long time to respond, we can shorten the 15 sec timeout to ensure users aren't waiting too long on that screen